src/metpy/plots/wx_symbols.py

```diff
@@ -18,6 +19,77 @@
                                                            'fonts/wx_symbols.ttf'))
 
 
+@exporter.export
+def wx_code_to_numeric(codes):
+    """Determine the numeric weather symbol value from METAR code text.
+
+    A robust method to identifies the numeric value for plotting the correct symbol from a
+    decoded METAR current weather group. The METAR codes should be strings with no missing
+    values or NaN strings (empty strings are okay).
+
+    For example, if from a Pandas Dataframe sfc_df.wxcodes.fillna('')
+
+    Parameters
+    ----------
+    codes : Array like containing string values of METAR weather codes
+
+    Returns
+    -------
+    array of numeric codes of current weather symbols from the wx_code_map for use in
+    plotting.
+    """
+    wx_sym_list = []
+    for s in codes:
+        if ' ' in s:
+            wxcode = s.split()[0]
+        else:
+            wxcode = s
+        if len(wxcode) > 4:
+            try:
+                wx_sym_list.append(wx_code_map[wxcode])
+            except Exception:
```

Comment:
I assume they only exception we're expecting here is `KeyError`? Let's catch that rather than `Exception`. Otherwise, this could swallow errors due to unrelated errors.

Reply:
Sure thing.